platform.md

```diff
@@ -171,29 +606,71 @@ A more thorough explanation is provided in the [Buildpack Interface Specificatio
 
 ### Environment
 
-User-provided environment variables intended for build and launch SHOULD NOT come from the same list.
```

Comment:
is this not true anymore?

Reply:
User-provided environment variables supplied at build-time, regardless of whether they are intended to configure build-time or launch-time behavior, all live under `<platform>/env` so I would say this is not true. I am not sure it ever was true? Unless the wording is just confusing and therefore I have misunderstood its meaning?

User-provided environment variables supplied at launch-time work via a completely different mechanism. You set them in the lifecycle environment rather than writing them to `<platform>/env`. If _that_ is what this sentence was trying to convey, it isn't apparent to me.

But it's a good question. I will add a description of `<platform>/env` here, because that seems like an important part of the interface to a platform developer and it is currently missing.

Reply:
> User-provided environment variables supplied at launch-time work via a completely different mechanism. You set them in the lifecycle environment rather than writing them to <platform>/env. If that is what this sentence was trying to convey, it isn't apparent to me.

This was the intent. In Cloud Foundry, the same user-provided env vars are applied during staging and launch. The above line is supposed to discourage platforms from doing that in the future.

Reply:
Is it worth capturing that?